app/Http/Controllers/API/v1/StatusTagController.php

```diff
@@ -0,0 +1,339 @@
+<?php
+
+
+namespace App\Http\Controllers\API\v1;
+
+use App\Enum\StatusVisibility;
+use App\Http\Controllers\Backend\Transport\StatusTagController as StatusTagBackend;
+use App\Http\Resources\StatusTagResource;
+use App\Models\Status;
+use App\Models\StatusTag;
+use Illuminate\Auth\Access\AuthorizationException;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Validator;
+use Illuminate\Validation\Rules\Enum;
+use Illuminate\Validation\ValidationException;
+use function auth;
+
+class StatusTagController extends Controller
+{
+
+    /**
+     * @OA\Get(
+     *      path="/statuses/{statusId}/tags",
+     *      operationId="getTagsForStatus",
+     *      tags={"Status"},
+     *      summary="Show all tags for a status which are visible for the current user",
+     *      description="Returns a collection of all visible tags for the given status, if user is authorized to see
+     *      it",
+     *      @OA\Parameter (
+     *          name="statusId",
+     *          in="path",
+     *          description="Status-ID",
+     *          example=1337,
+     *          @OA\Schema(type="integer")
+     *      ),
+     *      @OA\Response(
+     *          response=200,
+     *          description="successful operation",
+     *          @OA\JsonContent(
+     *              @OA\Property(property="data", type="array",
+     *                  @OA\Items(
+     *                      @OA\Property(
+     *                          property="key",
+     *                          type="string"
+     *                      ),
+     *                      @OA\Property(
+     *                          property="value",
+     *                          type="string"
+     *                      ),
+     *                      @OA\Property(
+     *                          property="visibility",
+     *                          type="integer"
+     *                      ),
+     *                  )
+     *              ),
+     *          )
+     *       ),
```

Comment:
Please use a virtual model here since this is bad practice. You're also not including the status visibility enum here. Users won't know what the meaning of 0,1,2,3 is.